### ISSUE TITLE
preview.sh not previewing files with nested path on Windows

### ISSUE THREAD

**Issue by zuqini** [junegunn/fzf.vim#1513]
- [x] I have fzf 0.30.0 or above
- [x] I have read through https://github.com/junegunn/fzf.vim/blob/master/README.md
- [x] I have read through https://github.com/junegunn/fzf/blob/master/README-VIM.md
- [x] I have read through the manual page of fzf (`man fzf`)
- [x] I have searched through the existing issues

## Info

- OS
    - [ ] Linux
    - [ ] Mac OS X
    - [x] Windows
    - [ ] Etc.
- Shell
    - [x] bash
    - [ ] zsh
    - [ ] fish

## Problem / Steps to reproduce
Preview window is not showing file content for nested paths that include backslashes (e.g. `after\ftplugin\lua.vim` as opposed to `lua.vim`.
![image](https://github.com/junegunn/fzf.vim/assets/5790854/ae5b152c-d60e-4931-b8b0-2f71f5dd34b5)
vs.
![image](https://github.com/junegunn/fzf.vim/assets/5790854/68a63e73-5418-4dfd-a51a-22d6c03a4c55)

It looks like it's caused by `bash` not interpreting the unescaped `\` from the path returned by `fzf.vim\bin\preview.sh`.

Reproduce directly with fzf by:
1. running `fzf --preview='bash {...}\.vim\plugged\fzf.vim\bin\preview.sh {}' --bind shift-up:preview-page-up,shift-down:preview-page-down`
2. select a file with a nested path

To further debug the problem, I logged the command returned by `preview.sh` and ran it in bash directly and saw that the backslash was swallowed:
```
> bash
 $ highlight -O ansi -l after\ftplugin\lua.vim || coderay after\ftplugin\lua.vim || rougify after\ftplugin\lua.vim || cat after\ftplugin\lua.vim
bash: highlight: command not found
bash: coderay: command not found
bash: rougify: command not found
cat: afterftpluginlua.vim: No such file or directory
```
If I escape the path, then it previews the file content successfully:
```
> bash
$ cat after//ftplugin//lua.vim
set expandtab
set shiftwidth=2
set softtabstop=2
set tabstop=2
```

## Environment
This is happening on a new Windows 11 installation:
```
OS Name	Microsoft Windows 11 Pro
Version	10.0.22621 Build 22621
```

The vim version shouldn't matter since I reproduced the issue directly with `preview.sh`, but here it is anyway in case it helps:
```
NVIM v0.9.4
Build type: RelWithDebInfo
LuaJIT 2.1.1696883897
```

The `bash.exe` points to the git bash installed from `scoop`: `C:\Users\{username}\scoop\apps\git\current\bin\bash.exe`
  
I should note that this did not happen prior to my Windows 11 re-install. I have no idea what changed that suddenly caused this issue. Appreciate any pointers!

**Comment by junegunn** [comment]
Do you experience the same problem with the following command?

```vim
:call fzf#run(fzf#wrap({'options': ['--preview', 'cat {}']}))
```

**Comment by zuqini** [comment]
Nope, that fixes it!

**Comment by zuqini** [comment]
Nope, that works!

**Comment by DMinsky** [comment]
Same here. I could confirm that following works with vim in bash terminal.
`:call fzf#run(fzf#wrap({'options': ['--preview', 'cat {}']}))`

But in gvim cat is unavailable, so still no preview there.

**Comment by DMinsky** [comment]
It seems I've found a workaround. Although it's not even a workaround, but rather an improvement of the default preview situation, and it's mentioned right in the README of fzf.vim 🫠.

You need to install [bat](https://github.com/sharkdp/bat). A tool that, as far as I understand, replaces cat and adds syntax highlighting in the preview window. I just unpacked the archive with the bat binary and added this path to my path. After that, the preview window began to display the contents of the selected files.

![Screenshot 2024-04-04 140544](https://github.com/junegunn/fzf.vim/assets/463426/71702438-0762-448e-8b30-3ee4752cbce3)

**Comment by mbilyanov** [comment]
How did you get the syntax highlighting to work in the preview window?

**Comment by DMinsky** [comment]
> How did you get the syntax highlighting to work in the preview window?

I literally did nothing except install bat. Just make sure the folder with the bat executable is available in the PATH so that fzf.vim can find it.

**Comment by junegunn** [comment]
Fixed in d91dd754a933e3da9797253f9b6d268bb85d0f4d.

And yeah, having bat would avoid the bug.
